framework/py/flwr/server/superlink/linkstate/linkstate_test.py

```diff
@@ -122,11 +122,53 @@ def test_get_all_run_ids_empty(self) -> None:
         state = self.state_factory()
 
         # Execute
-        run_ids = state.get_run_ids()
+        run_ids = state.get_run_ids(None)
 
         # Assert
         assert len(run_ids) == 0
 
+    def test_get_run_ids_with_flwr_aid(self) -> None:
+        """When a specific flwr_aid is passed, only its run_ids are returned."""
+        state = self.state_factory()
+
+        # Prepare - Create three runs with different flwr_aid values
+        run_id1 = state.create_run(None, None, "hash1", {}, ConfigRecord(), "userA")
+        run_id2 = state.create_run(None, None, "hash2", {}, ConfigRecord(), "userB")
+        run_id3 = state.create_run(None, None, "hash3", {}, ConfigRecord(), "userA")
+
+        # Execute - Only the runs for "userA" should be returned
+        result_userA = state.get_run_ids("userA")
+
+        # Assert
+        assert run_id1 in result_userA
+        assert run_id3 in result_userA
+        assert run_id2 not in result_userA
+        assert len(result_userA) == 2
```

Comment:
Could we do this? Similar to your code below.

```suggestion
        assert result_userA == {run_id1, run_id2}
```